### PR TITLE
SLE-747: IDE to server path matching failure

### PR DESCRIPTION
**Pull request by thahnen**
Handle the corner case in which a file with issues cannot be matched to server issues because the file itself does not comply with the server path.

### CODE REVIEW

File: org.sonarlint.eclipse.core.tests/src/test/java/org/sonarlint/eclipse/core/internal/tracking/ProjectIssueTrackerTest.java [hunk 1]
```diff
@@ -207,4 +230,17 @@ public void should_match_local_issues_by_text_range_hash() {
...
 
+  /** Corner case in which the IDE path cannot be converted to server path */
+  @Test
+  public void test_trackWithServerIssues_path_null() {
+    try {
+      underTest.trackWithServerIssues(new ProjectBinding("testProject", "/", "/"), List.of(file1), false, null);
+    } catch(NullPointerException ignored) {
+      // INFO: Ignore the failure on the job AFTERWARDS, this test only checks for the corner case!
+      // INFO: The NPE is coming from 'JobUtils.waitForFuture' because 'monitor' is null, not from the SLCORE method!
+    }
```
Comment (damien-urruty-sonarsource) [inline org.sonarlint.eclipse.core.tests/src/test/java/org/sonarlint/eclipse/core/internal/tracking/ProjectIssueTrackerTest.java:241]: Couldn't you create a dummy monitor instead of catching the NPE?

Comment (thahnen) [reply org.sonarlint.eclipse.core.tests/src/test/java/org/sonarlint/eclipse/core/internal/tracking/ProjectIssueTrackerTest.java:241]: I could, but then it fails with an NPE when running the Job at the point it is trying to get the `configScopeId` because I don't mock the resource and project of the `ISonarLintFile` in the test class.
"Failing fast", in this case, is the laziest but best option to test the corner case, in my opinion.

Comment (damien-urruty-sonarsource) [reply org.sonarlint.eclipse.core.tests/src/test/java/org/sonarlint/eclipse/core/internal/tracking/ProjectIssueTrackerTest.java:241]: IMHO it's never a good option to catch NPE, there is even a Sonar [rule](https://rules.sonarsource.com/java/RSPEC-1696/) about that, because it can potentially hide other problems. What if one day a regression is introduced, a "real" NPE is thrown elsewhere but silently caught here?
The usage of mocks is another smell to me, but that's a different debate, and it would imply rewriting the tests.
So I would suggest to find another solution, maybe relying on mocks

Comment (thahnen) [reply org.sonarlint.eclipse.core.tests/src/test/java/org/sonarlint/eclipse/core/internal/tracking/ProjectIssueTrackerTest.java]: The test is fixed using a mock
